webapi/Controllers/ContentModeratorController.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using CopilotChat.WebApi.Options;
+using CopilotChat.WebApi.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace CopilotChat.WebApi.Controllers;
+
+[ApiController]
+[Authorize]
+public class ContentModeratorController : ControllerBase
+{
+    private readonly ILogger<ContentModeratorController> _logger;
+    private readonly ContentModeratorOptions _options;
+    private readonly AzureContentModerator? _contentModerator = null;
+
+    /// <summary>
+    /// The constructor of ContentModeratorController.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="ContentModeratorOptions">The content moderation options.</param>
+    /// <param name="contentModerator">The content moderation service.</param>
+    public ContentModeratorController(
+        ILogger<ContentModeratorController> logger,
+        IOptions<ContentModeratorOptions> contentModeratorOptions,
+        AzureContentModerator? contentModerator = null)
+    {
+        this._logger = logger;
+        this._contentModerator = contentModerator;
+        this._options = contentModeratorOptions.Value;
+    }
+
+    /// <summary>
+    /// Detect sensitive image content.
+    /// </summary>
+    /// <param name="base64Image">The base64 encoded image for analysis.</param>
+    /// <returns>The HTTP action result.</returns>
+    [HttpPost]
+    [Route("contentModerator/image")]
```

Comment:
images should be plural in the route to align with REST best practices of resources and collections of resources.

Reply:
should we go further and make this route more generic (i.e. by changing the route to just `/contentModerator`)? I guess with that said, will there be other types of content we're moderating in the future?

Reply:
This is no longer relevant -- I removed this controller since we're not making any analysis requests from the client directly.